web3/utils/abi.py

```diff
@@ -1,21 +1,512 @@
+            arg_count_matches,
+            encoding_matches,
+            args,
+            kwargs,
+        )
+
+        raise MismatchedABI(error_diagnosis)
+
+    return _element_abi_to_type(function_matches[0])
+
+
+def get_receive_function_abi(contract_abi: ABI) -> ABIReceive:
+    """
+    Return the receive function ABI from the contract ABI.
+
+    :param contract_abi: Contract ABI.
+    :type contract_abi: `ABI`
+    :return: Receive function ABI.
+    :rtype: `ABIReceive`
+
+    .. doctest::
+
+        >>> from web3.utils.abi import get_receive_function_abi
+        >>> abi = [
+        ...     {
+        ...         "constant": False,
+        ...         "inputs": [],
+        ...         "name": "receive",
+        ...         "outputs": [],
+        ...         "payable": False,
+        ...         "stateMutability": "nonpayable",
+        ...         "type": "receive",
+        ...     }
+        ... ]
+        >>> get_receive_function_abi(abi)
+        {'constant': False, 'inputs': [], 'name': 'receive', 'outputs': [], \
+'payable': False, 'stateMutability': 'nonpayable', 'type': 'receive'}
+    """
+    receive_abis = filter_abi_by_type("receive", contract_abi)
+    if receive_abis and receive_abis[0]["type"] == "receive":
+        return cast(ABIReceive, receive_abis[0])
+    else:
+        raise ABIReceiveNotFound("No receive function was found in the contract ABI.")
+
+
+def get_fallback_function_abi(contract_abi: ABI) -> ABIFallback:
+    """
+    Return the fallback function ABI from the contract ABI.
+
+    :param contract_abi: Contract ABI.
+    :type contract_abi: `ABI`
+    :return: Fallback function ABI.
+    :rtype: `ABIFallback`
+
+    .. doctest::
+
+        >>> from web3.utils.abi import get_fallback_function_abi
+        >>> abi = [
+        ...     {
+        ...         "constant": False,
+        ...         "inputs": [],
+        ...         "name": "fallback",
+        ...         "outputs": [],
+        ...         "payable": False,
+        ...         "stateMutability": "nonpayable",
+        ...         "type": "fallback",
+        ...     }
+        ... ]
+        >>> get_fallback_function_abi(abi)
+        {'constant': False, 'inputs': [], 'name': 'fallback', 'outputs': [], \
+'payable': False, 'stateMutability': 'nonpayable', 'type': 'fallback'}
+    """
+    fallback_abis = filter_abi_by_type("fallback", contract_abi)
+    if fallback_abis and fallback_abis[0]["type"] == "fallback":
+        return cast(ABIFallback, fallback_abis[0])
+    else:
+        raise ABIFallbackNotFound("No fallback function was found in the contract ABI.")
+
+
+def check_if_arguments_can_be_encoded(
+    function_abi: ABIElement,
+    args: Sequence[Any],
+    kwargs: Dict[str, Any],
+    abi_codec: Optional[Any] = None,
+) -> bool:
+    """
+    Check if the provided arguments can be encoded with the function ABI.
+
+    :param function_abi: Function ABI.
+    :type function_abi: `ABIFunction`
```

Comment:
See comment above but I think this is the right change to make out of the two?

```suggestion
    :type abi_element: `ABIElement`
```